pkg/stanza/operator/helper/emitter.go

```diff
@@ -145,9 +145,17 @@ func (e *LogEmitter) flusher(ctx context.Context) {
 		select {
 		case <-ticker.C:
 			if oldBatch := e.makeNewBatch(); len(oldBatch) > 0 {
-				e.flush(ctx, oldBatch)
+				e.flush(context.Background(), oldBatch)
 			}
 		case <-ctx.Done():
+			// flush currently batched entries
+			for {
+				oldBatch := e.makeNewBatch()
+				if len(oldBatch) == 0 {
+					break
+				}
+				e.flush(context.Background(), oldBatch)
+			}
```

Comment:
IMO this is a bit more clear
```suggestion
			for oldBatch := e.makeNewBatch(); len(oldBatch) > 0; oldBatch = e.makeNewBatch() {
				e.flush(context.Background(), oldBatch)
			}
```

Reply:
i like it